src/ccs/multifolding.rs

```diff
@@ -25,58 +25,108 @@ use core::{cmp::max, marker::PhantomData};
 use ff::{Field, PrimeField};
 use flate2::{write::ZlibEncoder, Compression};
 use itertools::concat;
+use rand_core::RngCore;
 use rayon::prelude::*;
 use serde::{Deserialize, Serialize};
 use sha3::{Digest, Sha3_256};
 use std::ops::{Add, Mul};
 use std::sync::Arc;
 
-// XXX: THe idea is to have Multifolding as IVC instance in the future, holding the main CCS
-// instances. Then the rest of CCS, CCCS, LCCCS hold references to it.
-// Is our single source of data.
+/// The NIMFS structure is the center of operations of the folding scheme.
+/// Once generated, it allows us to fold any upcomming CCCS instances within it without needing to do much.
+// XXX: Pending to add doc examples.
```

Comment:
I'd add a comment on what NIMFS stands for and rationale for name vs NIFS (came up/will come up during PR review upstream)

Reply:
Also should we call this file `nimfs.rs` for consistency?

Reply:
I'd do it on a different PR. Otherwise the diff will look as all additions and make the changes less visible.

Reply:
Ok let's do in follow up PR so we don't forget it